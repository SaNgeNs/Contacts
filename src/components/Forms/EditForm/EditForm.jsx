import React, { memo } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import Button from '@material-ui/core/Button';
import SaveIcon from '@material-ui/icons/Save';
import { useDispatch } from 'react-redux';

import updateCard from 'Actions/updateCard';
import FieldInput from 'Components/FieldInput';
import validation from 'Utils/validation';

import styles from './styles';

export const EditForm = ({
  classes,
  handleSubmit,
  cardInfo = {},
  onClickAfterSubmit = () => {},
  initialValues = {
    name: '',
    phone: '',
    email: '',
  },
}) => {
  const dispatch = useDispatch();

  const submitForm = (values) => {
    validation(values);

    dispatch(updateCard({
      ...cardInfo,
      ...values,
    }));

    onClickAfterSubmit();
  };

  return (
    <form
      className={classes.form}
      noValidate
      onSubmit={handleSubmit(submitForm)}
    >
      <FieldInput name="name" id="editName" label="name" type="text" variant="outlined" />
      <FieldInput normalize={(val) => val.replace(/\D+/g,"").slice(0, 9)} name="phone" id="editPhone" label="phone" type="tel" variant="outlined" />
      <FieldInput name="email" id="editEmail" label="email" type="email" variant="outlined" />

      <Button
        type="submit"
        variant="contained"
        color="primary"
        size="large"
        startIcon={<SaveIcon />}
      >
        Save
      </Button>
    </form>
  );
};

export default memo(withStyles(styles)(EditForm));
