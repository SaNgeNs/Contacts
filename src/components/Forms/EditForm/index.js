import withErrorHandler from 'Components/withErrorHandler';
import loadable from '@loadable/component';
import { reduxForm } from 'redux-form';

const EditForm = loadable(() => import(/* webpackChunkName: "edit_form_component" */'./EditForm'));

export default withErrorHandler(reduxForm({
  form: 'editCard',
})(EditForm));
