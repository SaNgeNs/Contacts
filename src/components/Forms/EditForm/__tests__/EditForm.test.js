import React from 'react';
import { fireEvent } from '@testing-library/react';
import customRender from 'Mock/customRender';
import { reduxForm } from 'redux-form';

import EditForm from '../EditForm';

describe('<EditForm />', () => {
  it('should render edit form', () => {
    const Component = reduxForm({
      form: 'editCard',
    })(EditForm);

    const { container, unmount } = customRender(
      <Component
        initialValues={{
          name: '123',
          phone: '123123123',
          email: 'asdsd@masil.asd',
        }}
      />,
    );

    expect(container).toMatchSnapshot();
    unmount();
  });

  it('should called func onClickAfterSubmit: ', () => {
    const Component = reduxForm({
      form: 'editCard',
    })(EditForm);

    const mockFn = jest.fn();

    const { container, unmount } = customRender(
      <Component
        onClickAfterSubmit={mockFn}
        initialValues={{
          name: '123',
          phone: '123123123',
          email: 'asdsd@masil.asd',
        }}
      />,
    );

    const button = container.querySelector('[type="submit"]');
    fireEvent.click(button);

    expect(mockFn).toBeCalledTimes(1);
    unmount();
  });

  it('should return form with default props: ', () => {
    const Component = reduxForm({
      form: 'editCard',
    })(EditForm);

    const { container, unmount } = customRender(
      <Component />,
    );

    expect(container).toMatchSnapshot();
    unmount();
  });
});
