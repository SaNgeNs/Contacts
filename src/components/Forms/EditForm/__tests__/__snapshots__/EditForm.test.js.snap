// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditForm /> should render edit form 1`] = `
<div>
  <form
    class="EditForm-form-1"
    novalidate=""
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeLarge"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
            />
          </svg>
        </span>
        Save
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;

exports[`<EditForm /> should return form with default props:  1`] = `
<div>
  <form
    class="EditForm-form-3"
    novalidate=""
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
      tabindex="0"
      type="submit"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeLarge"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V7l-4-4zm-5 16c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3zm3-10H5V5h10v4z"
            />
          </svg>
        </span>
        Save
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
